test/cql-pytest/test_materialized_view.py

```diff
@@ -1189,3 +1190,68 @@ def test_base_partition_deletion_with_low_timestamp(cql, test_keyspace):
             # Verify we get only the row with the newer timestamp
             assert [2] == [x.c for x in cql.execute(f"SELECT c FROM {table} WHERE p=0")]
             assert [2] == [x.c for x in cql.execute(f"SELECT c FROM {mv} WHERE p=0")]
+
+# Verify that creating a materialized view with a specific ID respects it,
+# i.e. the created MV uses the ID indeed.
+def test_creating_mv_with_id(cql, test_keyspace):
+    with new_test_table(cql, test_keyspace, 'p int PRIMARY KEY') as table:
+        # For a given string of form `ks_name.cf_name`, returns `cf_name`.
+        def extract_cf_name(cf_full_name: str) -> str:
+            pattern = re.compile(r'\S+\.(\S+)')
+            matches = pattern.search(cf_full_name)
+            match = matches.group(1)
+            assert match is not None
+            return match
+
+        # Obtains a CQL statement that can be used to restore a given existing CF.
+        def extract_cf_restore_stmt(ks: str, type: str, cf: str) -> str:
+            desc_iter = cql.execute('DESC SCHEMA WITH INTERNALS')
+            desc_iter = filter(lambda row: row.keyspace_name == ks, desc_iter)
+            desc_iter = filter(lambda row: row.type == type, desc_iter)
+            desc_iter = filter(lambda row: row.name == cf, desc_iter)
+
+            # We expect to only get one row---one corresponding to restoring the CF.
+            [cf_row] = [*desc_iter]
+            return cf_row.create_statement
+
+        # Returns the ID value from a create statement corresponding to a CF.
+        def extract_id_value(create_statement: str) -> str:
+            pattern = re.compile(r'WITH ID = (\S+)')
+            matches = pattern.search(create_statement)
+            match = matches.group(1)
+            assert match is not None
+            return match
+
+        mv_name = None
+        mv_restore_stmt = None
+        mv_id = None
+
+        # Step 1. Create a materialized view, obtain a statement to restore it and its ID. Drop the MV.
```

Comment:
Interesting. You're basically mixing testing of "WITH ID =" with testing DESC MATERIALIZED VIEW (WITH INTERNALS?) and its ability to create a full materialized view - and not just extracting an ID and using it in your own new statement.
I think I can live with that.